src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -290,6 +293,23 @@ private Path path(final Key key) {
         return Paths.get(this.dir.toString(), key.string());
     }
 
+    /**
+     * Removes empty directories.
+     * @param key Key
+     */
+    private void deleteEmptyDirs(final Optional<Key> key) {
```

Comment:
@olenagerasimova I think we need to use `Path` parameter here, since the abstraction of file-system, path and directories is lower than storage and keys. When we're calling the method `deleteEmptyDirs`, where dir means directories, then we're using low-lever abstraction of the file-system and IO Java API

Reply:
@g4s8 I'd prefer to change the naming, not the parameter. When there are no parents of an empty one of the key, that means, that we reached the storage root folder.